conduit/pipeline.go

```diff
@@ -0,0 +1,286 @@
+			},
+			Type: "Importer",
+			Name: (*p.importer).Metadata().Name(),
+		},
+	)
+
+	// TODO modify/fix ?
+	jsonEncode := string(json.Encode(p.cfg.Importer["Config"]))
+	genesis, err := (*p.importer).Init(p.ctx, plugins.PluginConfig(jsonEncode), importerLogger)
+
+	var initProvider data.InitProvider = &PipelineInitProvider{
+		currentRound: 0,
+		genesis:      genesis,
+	}
+
+	p.initProvider = &initProvider
+
+	importerName := (*p.importer).Metadata().Name()
+	if err != nil {
+		return fmt.Errorf("could not initialize importer (%s): %w", importerName, err)
+	}
+	p.logger.Infof("Initialized Importer: %s", importerName)
+
+	for _, processor := range p.processors {
+
+		processorLogger := log.New()
+		processorLogger.SetFormatter(
+			PluginLogFormatter{
+				Formatter: &log.JSONFormatter{
+					DisableHTMLEscape: true,
+				},
+				Type: "Processor",
+				Name: (*processor).Metadata().Name(),
+			},
+		)
+
+		err := (*processor).Init(p.ctx, *p.initProvider, "")
+		processorName := (*processor).Metadata().Name()
+		if err != nil {
+			return fmt.Errorf("could not initialize processor (%s): %w", processorName, err)
+		}
+		p.logger.Infof("Initialized Processor: %s", processorName)
+
+	}
+
+	exporterLogger := log.New()
+	exporterLogger.SetFormatter(
+		PluginLogFormatter{
+			Formatter: &log.JSONFormatter{
+				DisableHTMLEscape: true,
+			},
+			Type: "Exporter",
+			Name: (*p.exporter).Metadata().Name(),
+		},
+	)
+
+	err = (*p.exporter).Init("", p.logger)
+	ExporterName := (*p.exporter).Metadata().Name()
+	if err != nil {
+		return fmt.Errorf("could not initialize Exporter (%s): %w", ExporterName, err)
+	}
+	p.logger.Infof("Initialized Exporter: %s", ExporterName)
+
+	return nil
+}
+
+func (p *pipelineImpl) Stop() error {
+	if err := (*p.importer).Close(); err != nil {
+		// Log and continue on closing the rest of the pipeline
+		p.logger.Errorf("Importer (%s) error on close: %v", (*p.importer).Metadata().Name(), err)
+	}
+
+	for _, processor := range p.processors {
+		if err := (*processor).Close(); err != nil {
+			// Log and continue on closing the rest of the pipeline
+			p.logger.Errorf("Processor (%s) error on close: %v", (*processor).Metadata().Name(), err)
+		}
+	}
+
+	if err := (*p.exporter).Close(); err != nil {
+		p.logger.Errorf("Exporter (%s) error on close: %v", (*p.exporter).Metadata().Name(), err)
+	}
+
+	return nil
+}
+
+// MakePipeline creates a Pipeline
+func MakePipeline(cfg *PipelineConfig, logger *log.Logger) (Pipeline, error) {
+
+	if cfg == nil {
+		return nil, fmt.Errorf("pipeline config was empty")
+	}
+
+	if err := cfg.Valid(); err != nil {
+		return nil, err
+	}
+
+	if logger == nil {
+		return nil, fmt.Errorf("logger was empty")
+	}
+
+	pipeline := &pipelineImpl{
+		ctx:          context.Background(),
+		cfg:          cfg,
+		logger:       logger,
+		initProvider: nil,
+		importer:     nil,
+		processors:   []*processors.Processor{},
+		exporter:     nil,
+	}
+
+	importerName, ok := cfg.Importer["Name"]
```

Comment:
Could you make this a property so that we don't need to check that it exists and cast it to a string?